pkg/reconciler/persistentvolumeclaim/pvc_role.go

```diff
@@ -0,0 +1,199 @@
+package persistentvolumeclaim
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	apiv1 "github.com/cloudnative-pg/cloudnative-pg/api/v1"
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/specs"
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/utils"
+)
+
+// PVCRole is the common interface for all PVC roles
+type PVCRole interface {
+	// GetLabels will be used as the label value
+	GetLabels(instanceName string) map[string]string
+	// GetPVCName will be used to get the name of the PVC
+	GetPVCName(instanceName string) string
+	// GetStorageConfiguration will return the storage configuration to be used
+	// for this PVC role and this cluster
+	GetStorageConfiguration(cluster *apiv1.Cluster) (apiv1.StorageConfiguration, error)
+	// GetSource gets the PVC source to be used when creating a new PVC
+	GetSource(source *StorageSource) (*corev1.TypedLocalObjectReference, error)
+	// GetRoleName return the role name in string
+	GetRoleName() string
+}
+
+// PVCRolePgData is a PVC used for storing PG_DATA
+var PVCRolePgData = &PgData{roleName: utils.PVCRoleValueData}
+
+// PVCRolePgWal is a PVC used for storing PG_WAL
+var PVCRolePgWal = &PgWal{roleName: utils.PVCRoleValueWal}
+
+// PVCRolePgTablespace is a PVC used for storing Tablespace
+var PVCRolePgTablespace = &PgTablespace{roleName: utils.PVCRoleValueTablespace}
+
+// NewPGTablespacePVCRole is a construct to get a tablespace PVC role
+func NewPGTablespacePVCRole(tbsName string) PVCRole {
+	return &PgTablespace{utils.PVCRoleValueTablespace, tbsName}
+}
+
+// GetPVCRole return pvcRole based on the roleName given
+func GetPVCRole(roleName string, tbsName string) (PVCRole, error) {
```

Comment:
We could have `labels map[string]string` as parameters instead of the role and the tablespace name